### ISSUE TITLE
Browser compatibility data on OPFS page

### ISSUE THREAD

**Issue by josephrocca** [mdn/content#37004]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/File_System_API/Origin_private_file_system

### What specific section or headline is this issue about?

Browser compatibility

### What information was incorrect, unhelpful, or incomplete?

At the top of the page there's some text "supporting browsers" which link to this URL: https://developer.mozilla.org/en-US/docs/Web/API/File_System_API/Origin_private_file_system#browser_compatibility

But there's no browser compat section on that page, so nothing happens when you click it.

### What did you expect to see?

Browser compat data

### Do you have anything more you want to share?

Thanks!

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/file_system_api/origin_private_file_system`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/File_System_API/Origin_private_file_system
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/file_system_api/origin_private_file_system/index.md
* Last commit: https://github.com/mdn/content/commit/f216422c99b6c7014e398803b70600501bce8a48
* Document last modified: 2024-10-16T05:12:45.000Z

</details>

**Comment by josephrocca** [comment]
Hey @Josh-Cena, this was originally about the broken link, because I though the compat data was on compiled in one spot on the File System Access API page, but then I realized that it's nowhere to be found, so I updated the issue to be about the compat data itself.

I notice that there are other issues here about lack of compat data which are open, so I assume this one should stay open?

I unfortunately don't have comprehensive info, but after some research just now, this is what I determined:

* Supported in Firefox since v111: https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Releases/111
* Supported in Safari since macOS 12.2 and iOS 15.2:
  * https://webkit.org/blog/12257/the-file-system-access-api-with-origin-private-file-system/
  * https://developer.apple.com/documentation/safari-release-notes/safari-15_2-release-notes
* Chrome Desktop since v102 and Chrome Android since v109: https://developer.chrome.com/blog/new-in-chrome-109

Notable quote from a popular database: https://rxdb.info/rx-storage-opfs.html

> From the beginning of 2023, the Origin Private File System API is supported by all modern browsers like Safari, Chrome, Edge and Firefox. Only Internet Explorer is not supported and likely will never get support.

**Comment by josephrocca** [comment]
@Josh-Cena AFAIK the above versions I gave are for `nagivator.storage.getDirectory`, since that's all you need for OPFS support. IIUC, everything else is specified "upstream" since it's used for regular File System Access API stuff.

**Comment by josephrocca** [comment]
Is this convention strict?

I noticed that this API landing page does have `xr` compatibility data:

https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API

Which is copied from here:

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/xr

I'm wondering how people are supposed to work out whether a browser supports OPFS. If analogy to WebXR example isn't possible, would it e.g. be possible to add a "Browser support" section at the bottom of the page which has a sentence or two that directs people to https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/getDirectory ? Or some other way to help people find it?

**Comment by josephrocca** [comment]
> WebXR, has a single entrypoint

That makes sense, but that's exactly the case we have here, right? I.e. 'OPFS' as a feature is really just `nagivator.storage.getDirectory` - and from there onwards it's basically just normal File System Access stuff.

I assume this is why, for example, web.dev links to the `nagivator.storage.getDirectory` page if you click "Source" their support data:

https://web.dev/articles/origin-private-file-system

![image](https://github.com/user-attachments/assets/8bfb057d-31ff-4d12-a976-cc7804d494f3)

**Comment by Josh-Cena** [comment]
That sounds reasonable—let's see what others say.

**Comment by wbamberg** [comment]
This isn't a landing page either though, it's a guide page, and we don't typically include banners like this in guide pages, because of problems like this. I think we should just remove both these banners.

**Comment by Josh-Cena** [comment]
Problem is that OPFS is more or less a standalone API surface in its own right. It just doesn't have its own spec and hence its own landing page, but I wouldn't call it related to, say, `showDirectoryPicker()`.

**Comment by wbamberg** [comment]
What I mean is, this isn't a reference page for a concrete web platform API. That would be the page for [`getDirectory()`](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager/getDirectory), which is a reference page for a Web API method, so it has compat, and rules derived from the IDL for things like like secure context and worker availability. This is really a conceptual page about what an OPFS is, what the restrictions of using it are, and so on. And because it's not a concrete web platform API, it doesn't have those things that the banners are for.

But, it's just my opinion.

**Comment by Josh-Cena** [comment]
Yeah. I understand these structural limitations, which I've said above. I'm saying that, for the purpose of readers, this page is not a guide of an API—it _is_ a formal reference for a standalone API and should actually be parallel with the https://developer.mozilla.org/en-US/docs/Web/API/File_System_API page, meaning readers will expect its own implementation status and its own banner. Just to say what readers expect; I don't have strong opinions either.

**Comment by KazimirPodolski** [comment]
Well, current state is obviously confusing to people.

![Image](https://github.com/user-attachments/assets/63899e15-4272-42fc-82c4-82253c19caf9)

**Comment by Josh-Cena** [comment]
I'm actually surprised that people are confused (though I agree with them and think a browser compat table should be present). Why don't we get issues for https://developer.mozilla.org/en-US/docs/Web/API/WebOTP_API, or https://developer.mozilla.org/en-US/docs/Web/API/WebUSB_API, or https://developer.mozilla.org/en-US/docs/Web/API/Payment_Request_API/Concepts? They are all guide or API pages that talk about "supporting browsers" in a banner but have no compat section.

**Comment by KazimirPodolski** [comment]
~Did all those have "supporting browsers" section before? That might be why you didn't get reports. You're not inclined to report what you don't even know could be here.~

I wonder tho if some kind of browser compat still can be introduced, or at least some shotcuts for related APIs. Like, "I just wanna know if I can use OPFS in 2025 or not". Also https://caniuse.com/?search=OPFS is completely empty which doesn't help the case either.

**Comment by Josh-Cena** [comment]
In that case you would look for one interface, like https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryHandle, and read its support instead. It doesn't make sense to talk about support for one entire API, because interfaces and methods can be continuously added to an API.

**Comment by Josh-Cena** [comment]
Fixed by https://github.com/mdn/content/pull/40331
